### ISSUE TITLE
Update `pandas` dependency range

### ISSUE THREAD

**Issue by csala** [sdv-dev/Copulas#266]
Copulas currently does not support `pandas>=1.1.5`, which does not include the newest versions.

We should investigate why blocks us from using the latest versions and adapt the code if necessary.

**Comment by fealho** [comment]
Currently, if we run `pytest` with `pandas > 1.1.5` three test cases fail. After some investigation, I believe they fail because of [this](https://github.com/pandas-dev/pandas/issues/37448) update on their side. This change modifies the behavior of the `corr()` function which we use here:

https://github.com/sdv-dev/Copulas/blob/8d7f4e3e9f2060c607080e65e3a8d5fa4194081b/tests/unit/multivariate/test_tree.py#L323

We then use the tau matrix in the fitting process:

https://github.com/sdv-dev/Copulas/blob/8d7f4e3e9f2060c607080e65e3a8d5fa4194081b/tests/unit/multivariate/test_tree.py#L333

Since the `pandas` update changes our tau matrix, which in turn changes the fitting procedure, we end up generating a different tree in the end. Since the test cases expect the tree generated by the old `pandas`, they fail the current test.

In summary, after we correct the test cases that fail, I believe there will be no problems in updating `pandas` for all our libraries.

**Comment by rivendellx** [comment]
> 

For my other project, I must use pandas 0.25.4. just wondering would it cause any issues in the correlation/covariance/tau_mat matrix produced by the newest copulas package?
